realm/src/main/java/io/realm/Realm.java

```diff
@@ -656,6 +687,15 @@ private static void validateAgainstExistingConfigurations(RealmConfiguration new
                 throw new IllegalArgumentException("Two configurations with different schemas are trying to open " +
                         "the same Realm file. Their schema must be the same: " + newConfiguration.getPath());
             }
+
+            // Check if the durability is the same
+            SharedGroup.Durability cachedDurability = cachedConfiguration.getDurability();
+            SharedGroup.Durability newDurability = newConfiguration.getDurability();
+            if (cachedDurability.value != newDurability.value) {
+                throw new IllegalArgumentException("Configurations cannot have different durabilities for the " +
+                        "same realm filename/identifier: " + newConfiguration.getRealmFileName() + " " +
```

Comment:
identifier is not a word we have used before. Until we add more durability settings I would rephrase this to something like "A Realm cannot be both in-memory and persisted. Two conflicting configurations pointing to <filename> are being used."